config/src/config/node_config.rs

```diff
@@ -286,4 +459,114 @@ mod test {
         let merged_node_config = merge_node_config(node_config, override_node_config);
         assert!(matches!(merged_node_config, Err(Error::Unexpected(_))));
     }
+
+    #[test]
+    fn test_override_node_config_no_diff() {
+        let override_config = OverrideNodeConfig::new(NodeConfig::default(), NodeConfig::default());
+        let diff_yaml = override_config.get_yaml().unwrap();
+        assert_eq!(diff_yaml, serde_yaml::Value::Null);
+    }
+
+    #[test]
+    fn test_override_node_config_with_bool() {
+        let mut override_config =
+            OverrideNodeConfig::new(NodeConfig::default(), NodeConfig::default());
+        let config = override_config.config_mut();
+        config.api.enabled = false;
+
+        let diff_yaml = override_config.get_yaml().unwrap();
+        let expected_yaml: serde_yaml::Value = serde_yaml::from_str(
+            r#"
+                api:
+                    enabled: false
+                "#,
+        )
+        .unwrap();
+        assert_eq!(diff_yaml, expected_yaml);
+    }
+
+    #[test]
+    fn test_override_node_config_with_enum() {
+        let mut override_config =
+            OverrideNodeConfig::new(NodeConfig::default(), NodeConfig::default());
+        let config = override_config.config_mut();
+        config.base.waypoint = WaypointConfig::FromFile(PathBuf::from("test"));
+        let diff_yaml = override_config.get_yaml().unwrap();
+        let expected_yaml: serde_yaml::Value = serde_yaml::from_str(
+            r#"
+                base:
+                    waypoint:
+                        from_file: test
+                "#,
+        )
+        .unwrap();
+        assert_eq!(diff_yaml, expected_yaml);
+    }
+
+    #[test]
+    fn test_override_node_config_with_empty_base_vector() {
+        let mut override_config =
+            OverrideNodeConfig::new(NodeConfig::default(), NodeConfig::default());
+        let config = override_config.config_mut();
+        config.full_node_networks.push(Default::default());
+        config.full_node_networks.push(Default::default());
+        let diff_yaml = override_config.get_yaml().unwrap();
+        let default_node_config = serde_yaml::to_value(&NetworkConfig::default()).unwrap();
+        let mut expected_yaml: serde_yaml::Value = serde_yaml::Value::Null;
+        expected_yaml["full_node_networks"] =
+            serde_yaml::Value::Sequence(vec![default_node_config.clone(), default_node_config]);
+        // TODO: why don't the actual Values match, only matches with as_str?
```

Comment:
nit: it might be worth creating a helper function for this, e.g., `assert_equal_yaml()`? That way the"ugliness" and TODO is only in one place 😄